src/main/java/com/icthh/xm/ms/entity/repository/search/elasticsearch/index/ElasticSingleIndexNameResolver.java

```diff
@@ -0,0 +1,22 @@
+package com.icthh.xm.ms.entity.repository.search.elasticsearch.index;
+
+import static com.icthh.xm.ms.entity.config.elasticsearch.CustomMappingElasticsearchEntityInformation.XMENTITY_SUFFIX;
+
+import com.icthh.xm.ms.entity.config.elasticsearch.ElasticsearchConfiguration;
+import lombok.RequiredArgsConstructor;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConditionalOnProperty(value = "application.use-elasticsearch-multiple-indices",
+    havingValue = "false", matchIfMissing = true)
+@RequiredArgsConstructor
+public class ElasticSingleIndexNameResolver implements ElasticIndexNameResolver {
+
+    private final ElasticsearchConfiguration.IndexName indexName;
+
+    @Override
+    public String resolve(String typeKey) {
+        return indexName.getPrefix() + XMENTITY_SUFFIX;
```

Comment:
please check backward compatibility for all way define index name, and add test for it

Reply:
as of now the index name is consist from prefix and hard-coded  "xmentity"
we can see it in `XmEntity `class `@Document(indexName = "#{@indexName.prefix}xmentity")`
so all must be good. And i'll add a test

Reply:
in some deployment we specify index prefix name from application.yml
and pls check multi tenancy

Reply:
> in some deployment we specify index prefix name from application.yml
o_O how come ?  `indexName.prefix  = newIndexName`  ? Does spring so much smart ? 
I can see  that u can specify `ApplicationProperties.elasticSchemaSuffix` but i do not understand how u can replace indexName.prefix
>and pls check multi tenancy
A method `getPrefix()` builds prefix from tenant and elasticSchemaSuffix. So, think, multi tenancy  does work. 
But maybe i do not understand how u changeindex.prefix from config

Reply:
@sergeysenja1992 pls comment